pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -188,7 +188,9 @@ func separateVSHostsAndServices(virtualService config.Config,
 				// This service doesn't have a virtualService that matches it.
 				continue
 			}
-			foundSvcMatch = true // we did find a match
+			// If this service host matches the virtual service host at all, then it's a match
+			// and we shouldn't add it to the list of hosts to prevent duplicates
+			foundSvcMatch = svcHost.Matches(hostname)
```

Comment:
why do we need to match again

Reply:
If the wildcard VS host name matches the host name of a service, we need to make sure it doesn't get added to eventual RDS config as an unknown service. By using matches here, I was trying to keep similar behavior to before the bug fix. If you have other suggestions, let me know!

Reply:
can you provide a valid  test case, i donot fully understand the case. And this function is already very confusing

Reply:
Agreed; I've had to read through this section of code 5 or 6 times to finally understand (what I think are) all of the pieces. Let me take another pass through and make some comments and potentially refactor

Reply:
I also feel weired that port 80 is special to have non serviceregistry host added to the route.

Reply:
Here is a link to the special port 80 logic; this is why my change caused the original issue: https://github.com/istio/istio/blob/ac901c3ed1a2455705709bd5e81df781d7a63083/pilot/pkg/networking/core/v1alpha3/route/route.go#L264

I can add a non serviceRegistry host name to the preexisting wildcard test case to prove that this only happens for port 80